swarm/src/lib.rs

```diff
@@ -1067,7 +1067,7 @@ where
                         let mut parameters = SwarmPollParameters {
                             local_peer_id: &this.local_peer_id,
                             supported_protocols: &this.supported_protocols,
-                            listened_addrs: &this.listened_addrs,
+                            listened_addrs: this.listened_addrs.values().flatten().collect(),
```

Comment:
I think you may be able to get away with returning a `Flatten<Values<'a>>` iterator here?

Reply:
`SwarmPollParamters` implements `PollParameters`, which requires for `PollParameters::ListenedAddressesIter` to implement `ExactSizeIterator`, which unfortunately is not implemented for `Flatten`.

Reply:
Damn, thanks for the explanation.